matrix/matrix.py

```diff
@@ -0,0 +1,184 @@
+"""Matrix data type class."""
+
+import operator
+from collections.abc import Iterable
+
+
+class Matrix:
+    """An immutable object representing a matrix data-type.
+
+    Attributes:
+        _matrix (tuple): contains the data about the matrix. Each line of
+            the matrix is represented by a tuple.
+    Note:
+        This implementation only supports squared matrices (n * n).
+    """
+
+    def __init__(self, *lines):
```

Comment:
In this case, I don't like the usage of *args
Just get the tuple as an argument